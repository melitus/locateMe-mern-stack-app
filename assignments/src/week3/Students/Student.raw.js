const React = require('react');

const afolabi = React.createElement('li', { id: 'afolabi_abiodun', key: 'afolabi_abiodun' }, 'Afolabi Abiodun');
const kennedy = React.createElement('li', { id: 'kennedy_ejike', key: 'kennedy_ejike' }, 'Kennedy Ejike');
const kenny = React.createElement('li', { id: 'kenny_emmanuel', key: 'kenny_emmanuel' }, 'Kenny Emmanuel');
const micah = React.createElement('li', { id: 'micah_joel', key: 'micah_joel' }, 'Micah Joel');
const okiki = React.createElement('li', { id: 'okiki_oye-lawrence', key: 'okiki_oye-lawrence' }, 'Okiki Oye-Lawrence');
const sunday = React.createElement('li', { id: 'sunday_aroh', key: 'sunday_aroh' }, 'Sunday Aroh');

const reactFragment = [afolabi, kennedy, kenny, micah, okiki, sunday];
 // console.log(reactElement)
const Students = React.createElement(
   'ul',
   { className: 'myList' },
   reactFragment
 );
console.log(Students);
export default Students;
