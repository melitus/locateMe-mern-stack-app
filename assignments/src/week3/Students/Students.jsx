import * as React from 'react';
import PropTypes from 'prop-types';

import Student from './Student/Student';

class Students extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      students: props.students
    };
  }

  render() {
    const { students } = this.props;

    return <ul>
      {students.map(({ firstName, lastName, title, summary }) => (
        <Student
          key={firstName + lastName}
          firstName={firstName}
          lastName={lastName}
          jobTitle={title}
          summary={summary}
        />
    ))}</ul> || null;
  }
}

Students.propTypes = {
  students: PropTypes.arrayOf(
    PropTypes.object
  ).isRequired,
};

Students.defaultProps = {
  students: [],
};

export default Students;
