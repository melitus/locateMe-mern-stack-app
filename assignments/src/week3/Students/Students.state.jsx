import { connect } from 'react-redux';

import Students from './Students';
// import fetchStudents from '../../client/redux/features/students/students';

const mapStateToProps = state => ({
  students: state,
});

const mapDispatchToProps =  dispatch => ({
  // fetchStudents
});

export default connect(mapStateToProps, mapDispatchToProps)(Students);
