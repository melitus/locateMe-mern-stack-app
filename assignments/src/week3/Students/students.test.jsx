import * as React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';
import Students from './Student/Student';

enzyme.configure({ adapter: new Adapter() });
describe('<Students />', () => {
  it('should render students', () => {
    const wrapper = shallow(
      <Students />
   );
    expect(wrapper).toMatchSnapshot();
  });
});
