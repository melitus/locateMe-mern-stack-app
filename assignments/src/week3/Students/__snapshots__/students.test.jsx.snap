exports[`<Students /> should render students 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Student />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="insert profile"
          src={undefined} />,
        <p>
          FirstName: 
           LastName: 
</p>,
        <p>
          JobTitle: 
</p>,
        <p>
          Summary: 
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "insert profile",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "FirstName: ",
            undefined,
            " LastName: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "FirstName: ",
          undefined,
          " LastName: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "JobTitle: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "JobTitle: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Summary: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Summary: ",
          undefined,
        ],
        "type": "p",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="insert profile"
            src={undefined} />,
          <p>
            FirstName: 
             LastName: 
</p>,
          <p>
            JobTitle: 
</p>,
          <p>
            Summary: 
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "insert profile",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "FirstName: ",
              undefined,
              " LastName: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "FirstName: ",
            undefined,
            " LastName: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "JobTitle: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "JobTitle: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Summary: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Summary: ",
            undefined,
          ],
          "type": "p",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
